numpyro/distributions/continuous.py

```diff
@@ -67,6 +66,147 @@
 )
 
 
+class AsymmetricLaplace(Distribution):
+    arg_constraints = {
+        "loc": constraints.real,
+        "scale": constraints.positive,
+        "asymmetry": constraints.positive,
+    }
+    reparametrized_params = ["loc", "scale"]
+    support = constraints.real
+
+    def __init__(self, loc=0.0, scale=1.0, asymmetry=1.0, validate_args=None):
+        self.loc, self.scale, self.asymmetry = promote_shapes(loc, scale, asymmetry)
+        batch_shape = lax.broadcast_shapes(
+            jnp.shape(loc), jnp.shape(scale), jnp.shape(asymmetry)
+        )
+        super(AsymmetricLaplace, self).__init__(
+            batch_shape=batch_shape, validate_args=validate_args
+        )
+
+    @lazy_property
+    def left_scale(self):
+        return self.scale * self.asymmetry
+
+    @lazy_property
+    def right_scale(self):
+        return self.scale / self.asymmetry
+
+    def log_prob(self, value):
+        if self._validate_args:
+            self._validate_sample(value)
+        z = value - self.loc
+        z = -jnp.abs(z) / jnp.where(z < 0, self.left_scale, self.right_scale)
+        return z - jnp.log(self.left_scale + self.right_scale)
+
+    def sample(self, key, sample_shape=()):
+        assert is_prng_key(key)
+        shape = (2,) + sample_shape + self.batch_shape + self.event_shape
+        u, v = random.exponential(key, shape=shape)
+        return self.loc - self.left_scale * u + self.right_scale * v
+
+    @property
+    def mean(self):
+        total_scale = self.left_scale + self.right_scale
+        return self.loc + (self.right_scale**2 - self.left_scale**2) / total_scale
+
+    @property
+    def variance(self):
+        left = self.left_scale
+        right = self.right_scale
+        total = left + right
+        p = left / total
+        q = right / total
+        return p * left**2 + q * right**2 + p * q * total**2
+
+    def cdf(self, value):
+        return jnp.where(
+            value <= self.loc,
+            self.asymmetry**2
+            / (1 + self.asymmetry**2)
+            * jnp.exp((self.scale / self.asymmetry) * (value - self.loc)),
+            1
+            - 1
+            / (1 + self.asymmetry**2)
+            * jnp.exp(-self.scale * self.asymmetry * (value - self.loc)),
+        )
+
+
+class AsymmetricLaplaceQuantile(Distribution):
```

Comment:
Maybe we can add docstrings to clarify the difference between two distributions? Something like given p follows AsymmetricLaplace and q follows AsymmetricLaplaceQuantile, p = q if
```
# assume without miscalculation
q.loc = p.loc, q.scale = p.scale * q.quantile / p.asymmetry, q.quantile = 1 / (1 + 1 / p.asymmetry ** 2)
```
and mention that in this distribution, quantile `quantile` of this distribution is `loc` (not sure how to describe this properly).

Reply:
I'm not sure there is an intuitive translation between the two parameterizations, but can play around with sympy for a bit to see if there's a straightforward conversion of `quantile` into `asymmetry`. In both distributions, `location` and `scale` have the same usual intent. Both parameterizations are represented as a mixture of two exponential distributions, and `asymmetry` and `quantile` are just two different ways of defining how much probability density occurs under the left vs. right components of the mixture. 

At the very least, I think it would make sense to add a docstring to the `AsymmetricLaplaceQuantile` class to describe that this class is intended for Bayesian quantile regression, and that quantile refers to the desired quantile of the data against which the model is being fit.

Reply:
The scale parameters in two distributions are a bit difference. With AsymLaplace(0,1,1) is Laplace(0,1) while AsymLaplaceQuantile(0,1,0.5) is Laplace(0,2). Maybe it is also worth to mention that to avoid confusions.

Reply:
Looking at your sample method, we have
```
q.right_scale = q.scale / q.quantile = p.scale / p.asymetry = p.right_scale
q.left_scale = q.scale / (1 - q.quantile) = q.scale / q.quantile * q.quantile / (1 - q.quantile)
= (using the relation of quantile and asymmetry in my last comment)
= p.scale / p.asymetry * p.asymmetry ** 2 = p.left_scale
```
Hope it help.

Reply:
Ok. Given the math you've done to show how to translate q (ALDQ) to p (ALD), I was able to redefine `AsymmetricLaplaceQuantile` as a child class of `AsymmetricLaplace`, calculating the conversion to `p.scale` from `q.scale` and calculating `p.asymmetry` from `q.scale` and `q.quantile`. 

I've also figured out the equation for `icdf` method for ALD, and have added that. Under the hood, ALDQ now just re-parameterizes the user input to use standard ALD parameterization, so no redundant code reproducing any of ALD's methods.